### DISCUSSION TITLE
Resource's Tables columns function to take also a Closure as argument

### DISCUSSION THREAD

**Discussion by invaders-xx** [filamentphp/filament#3523] [Help]
At the moment the columns function only accepts an array as argument. Will you consider an update to accepts also a closure ?

The situation is related to a RelationManager where the table's column should be build according to the ownerRecord configuration and as the RelationManager's table function is static, I didn't find a way to manage that except using a Closure with $livewire as parameter.

Any thoughts ?

David

**Comment by danharrin** [comment]
Which methods specifically? Some definitely do accept closures

**Reply by invaders-xx** [reply]
In Filament\Resources\Table, 

```php
public function columns(array $columns): static
    {
        $this->columns = $columns;

        return $this;
    }
```

**Reply by danharrin** [reply]
Instead of using `table()`, maybe override the native table builder `getTableColumns()` method, where you can access `$this->ownerRecord`
